and-cli-deploy-jenkins.js

```diff
@@ -0,0 +1,185 @@
+    const CREATE_PROFILE_OPTION = optionStringFactory.build("new");
+
+    // #endregion Constants
+
+    // -----------------------------------------------------------------------------------------
+    // #region Variables
+    // -----------------------------------------------------------------------------------------
+
+    let tag = null;
+    let job = null;
+
+    // #endregion Variables
+
+    // -----------------------------------------------------------------------------------------
+    // #region Functions
+    // -----------------------------------------------------------------------------------------
+
+    const jenkinsDeploy = {
+        async build() {
+            const userPrompt = prompt.getPrompt();
+            let profile = program.profile;
+            if (StringUtils.isEmpty(program.profile)) {
+                echo.error("Please specify a profile [--profile] to trigger");
+                shell.exit(1);
+            }
+            const config = jenkins.getConfig();
+            if (config.profiles.indexof(profile) < 0) {
+                echo.error(
+                    "Profile not found, please check your configuration"
+                );
+                shell.exit(1);
+            }
+
+            job = config.profiles[profile];
+            tag = await userPrompt.questionAsync("Tag name to deploy: ");
+            this.validateOrExit();
+
+            await prompt.confirmOrExit(
+                `Please confirm deployment of tag ${tag} to ${job}`
+            );
+
+            echo.message(`Jenkins starting ${job}, deploying version ${tag}`);
+            const url = `${config.url}/job/${job}/buildWithParameters?version=${tag}`;
+
+            const cmd = commandStringFactory.build(
+                "curl",
+                "-X",
+                "POST",
+                "-L",
+                `--user ${config.username}:${config.token}`,
+                url
+            );
+
+            if (shell.exec(cmd).code !== 0) {
+                echo.error(" - Failed to deploy jenkins");
+                shell.exit(1);
+            }
+        },
+        async createProfile() {
+            const userPrompt = prompt.getPrompt();
+            const jenkinsJobName = await userPrompt.questionAsync(
+                "Jenkins Job Name: "
+            );
+            const profileName = await userPrompt.questionAsync(
+                "Profile name (Nickname): "
+            );
+
+            let config = jenkins.getConfig();
+            let profiles = config.profiles;
+
+            if (profiles == null) {
+                profiles = Object.create(null);
+            }
+            profiles[profileName] = jenkinsJobName;
+            config.profiles = profiles;
+
+            jenkins.writeToConfig(config);
+            shell.exit(0);
+        },
+        async init() {
+            const configPath = jenkins.getConfigPath();
+            const userPrompt = prompt.getPrompt();
+
+            file.deleteIfExists(configPath);
```

Comment:
Thought - might be worth noting somewhere that this `init` option will overwrite an existing configuration file it one exists. 

To be honest, it might be better to short-circuit with an informational message if the config already exists. If I didn't remember that I had a config setup, I'd rather try to set one up and be told that it's already good to go instead of having it wipe out my stuff and then having to fill it back in.